File: src/Flagrum.Core/Graphics/Models/GameModelPart.cs
using Flagrum.Core.Serialization.MessagePack;

namespace Flagrum.Core.Graphics.Models;

public class GameModelPart : IMessagePackItem
{
    public string Name { get; set; }
    public uint Id { get; set; }
    public string Unknown { get; set; }
    public bool Flags { get; set; }

    public void Read(MessagePackReader reader)
    {
        Name = reader.Read<string>();
        Id = reader.Read<uint>();
        Unknown = reader.Read<string>();
        Flags = reader.Read<bool>();
    }

    public void Write(MessagePackWriter writer)
    {
        writer.Write(Name);
        writer.Write(Id);
        writer.Write(Unknown);
        writer.Write(Flags);
    }
}

File: src/Flagrum.Blender/core/graphics/models/gamemodelpart.py
from dataclasses import dataclass

from ...serialization.messagepack.reader import MessagePackReader
from ...serialization.messagepack.writer import MessagePackWriter


@dataclass(init=False)
class GameModelPart:
    name: str
    id: int
    unknown: str
    flags: bool

    def read(self, reader: MessagePackReader):
        self.name = reader.read()
        self.id = reader.read()
        self.unknown = reader.read()
        self.flags = reader.read()

    def write(self, writer: MessagePackWriter):
        writer.write(self.name)
        writer.write(self.id)
        writer.write(self.unknown)
        writer.write(self.flags)
